libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -172,6 +167,10 @@ void AC_WPNav::wp_and_spline_init(float speed_cms)
     _origin = _destination = stopping_point;
     _terrain_alt = false;
     _this_leg_is_spline = false;
+
+    // initialise the terrain velocity to the current maximum velocity
+    _terain_vel = _wp_speed_cms;
```

Comment:
Instead of _wp_speed_cms, I think this should be _wp_desired_speed_xy_cms which is set a bit higher up in the function.